tests/diagnostics/test_precipitation.py

```diff
@@ -0,0 +1,38 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import torch
+import pytest
+from earth2mip.diagnostic import PrecipitationAFNO
+
+
+@pytest.mark.slow
+@pytest.mark.xfail
```

Comment:
As with ClimateNet. Instantiate and test the class directly (w/o loading from checkpoint). we don't need to download the checkpoint to cover the PrecipitationAFNO code or do this minimal size based check...

likewise test at reduced resolution (e.g. 4x8)

Reply:
Like wise comment above, also grid resolution is baked into the diagnostic object itself (hard coded) since it reflects the checkpoint in the package not a general architecture.